kore/src/Kore/Builtin/Map.hs

```diff
@@ -460,6 +463,39 @@ asTermLike builtin =
     element (key, Domain.Value value) =
         mkApplySymbol elementSymbol [key, value]
 
+{- | Convert a Map-sorted 'TermLike' to its internal representation.
+
+The 'TermLike' is unmodified if it is not Map-sorted. @internalize@ only
+operates at the top-most level, it does not descend into the 'TermLike' to
+internalize subterms.
```

Comment:
This seems a bit hard to understand. By default I would assume that, if I'm trying to normalize `concat(alpha, concat(beta, gamma))`, then this won't descend into the `concat(beta, gamma)` subterm, it will only normalize the first concat, which is at the top-most level.

However, my guess is that this will descend into the inner `concat` to normalize.